tests/integration/accounts/test_contract_account.py

```diff
@@ -91,3 +93,25 @@ async def test_should_give_infinite_allowance_to_kakarot(
                 )
                 == "Uint256(low=340282366920938463463374607431768211455, high=340282366920938463463374607431768211455)"
             )
+
+        async def test_should_take_deployment_fees(
+            self, kakarot, eth, fund_evm_address
+        ):
+            evm_address = int(generate_random_evm_address(), 16)
+            computed_starknet_address = (
+                await kakarot.compute_starknet_address(evm_address).call()
+            ).result[0]
+
+            await fund_evm_address(evm_address)
+
+            await kakarot.deploy_externally_owned_account(evm_address).execute(
+                caller_address=4
+            )
+
+            # asserting that the balance of the account is the amount minus the deployment fee
+            assert (
+                await eth.balanceOf(computed_starknet_address).call()
+            ).result.balance.low == ACCOUNT_BALANCE
+
+            # asserting that the deployer gets the Deployment Fee back { initial balance of the deployer is 0 }
```

Comment:
you should compare balance of deployer before and after and not assume that it was 0

Reply:
Done.